File: go/fibonacci.go
package main

func FibonacciIter(position int) int {
	a, b := 1, 1

	for i := 1; i < position; i++ {
		b, a = a+b, b
	}
	return a
}

func FibonacciRecur(position int) int {

	if position == 1 || position == 2 {
		return 1
	}
	return FibonacciRecur(position-2) + FibonacciRecur(position-1)
}

File: js/fibonacci.ts
function fibonacciRecur(position: number) {

    if (position === 1 || position === 2) {
        return 1;

    }
    return fibonacciRecur(position-1) + fibonacciRecur(position-2);
}


function fibonacciIter(position: number) {
    let a = 1;
    let b = 1;
    let next = 0;
    for(let i = 1; i <position; i++){
        next = a + b
        a = b;
        b = next;
    }
    return a;
}
(()=> {
    console.log(fibonacciRecur(2));
    console.log(fibonacciRecur(10));
    console.log(fibonacciIter(2));
    console.log(fibonacciIter(10));
})()
